ivc/src/poseidon_8_56_5_3_2/mod.rs

```diff
@@ -0,0 +1,162 @@
+//! Specialised circuit for Poseidon where we have maximum degree 2 constraints.
+
+pub mod columns;
+pub mod interpreter;
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        poseidon_8_56_5_3_2::{columns::PoseidonColumn, interpreter, interpreter::PoseidonParams},
+        poseidon_params_8_56_5_3,
+        poseidon_params_8_56_5_3::PlonkSpongeConstantsIVC,
+    };
+    use ark_ff::UniformRand;
+    use kimchi_msm::{
+        circuit_design::{ColAccessCap, ConstraintBuilderEnv, WitnessBuilderEnv},
+        columns::ColumnIndexer,
+        lookups::DummyLookupTable,
+        Fp,
+    };
+    use mina_poseidon::permutation::poseidon_block_cipher;
+
+    pub struct PoseidonBN254Parameters;
+
+    pub const STATE_SIZE: usize = 3;
+    pub const NB_FULL_ROUND: usize = 8;
+    pub const NB_PARTIAL_ROUND: usize = 56;
+    pub const NB_TOTAL_ROUND: usize = NB_FULL_ROUND + NB_PARTIAL_ROUND;
+
+    type TestPoseidonColumn = PoseidonColumn<STATE_SIZE, NB_FULL_ROUND, NB_PARTIAL_ROUND>;
+    pub const N_COL: usize = TestPoseidonColumn::N_COL;
+    pub const N_DSEL: usize = 0;
+
+    impl PoseidonParams<Fp, STATE_SIZE, NB_TOTAL_ROUND> for PoseidonBN254Parameters {
+        fn constants(&self) -> [[Fp; STATE_SIZE]; NB_TOTAL_ROUND] {
+            let rc = &poseidon_params_8_56_5_3::static_params().round_constants;
+            std::array::from_fn(|i| std::array::from_fn(|j| Fp::from(rc[i][j])))
+        }
+
+        fn mds(&self) -> [[Fp; STATE_SIZE]; STATE_SIZE] {
+            let mds = &poseidon_params_8_56_5_3::static_params().mds;
+            std::array::from_fn(|i| std::array::from_fn(|j| Fp::from(mds[i][j])))
+        }
+    }
+
+    type PoseidonWitnessBuilderEnv = WitnessBuilderEnv<
+        Fp,
+        TestPoseidonColumn,
+        { <TestPoseidonColumn as ColumnIndexer>::N_COL },
+        { <TestPoseidonColumn as ColumnIndexer>::N_COL },
+        0,
+        0,
+        DummyLookupTable,
+    >;
+
+    #[test]
+    #[ignore = "There is no support for partial rounds in the crate poseidon"]
```

Comment:
Do I understand correctly that this test is disabled because you have no specification-level native implementation for this type of poseidon? 

Maybe the test is worth deleting though?

Reply:
See https://github.com/o1-labs/proof-systems/pull/2329 and https://github.com/o1-labs/proof-systems/pull/2328